src/Broadway/EventStore/DBALEventStore.php

```diff
@@ -170,4 +194,30 @@ private function deserializeEvent($row)
             DateTime::fromString($row['recorded_on'])
         );
     }
+
+    private function extractId($id)
+    {
+        if ($this->useBinary) {
+            try {
+                return Uuid::fromString($id)->getBytes();
+            } catch (\Exception $e) {
```

Comment:
@asm89 Well, we could force the usage of UUID for the binary mode and throw a `InvalidIdentifierException` or somthing along this line. What do you think?